src/v/cloud_storage/remote_segment.cc

```diff
@@ -77,64 +77,52 @@ remote_segment::remote_segment(
   cache& c,
   s3::bucket_name bucket,
   const manifest& m,
-  manifest::key path,
+  const manifest::key& name,
   retry_chain_node& parent)
   : _api(r)
   , _cache(c)
   , _bucket(std::move(bucket))
-  , _manifest(m)
-  , _path(std::move(path))
+  , _ntp(m.get_ntp())
   , _rtc(&parent)
   , _ctxlog(cst_log, _rtc, get_ntp().path())
   , _wait_list(expiry_handler_impl) {
+    auto meta = m.get(name);
+    vassert(meta, "Can't find segment metadata in manifest, name: {}", name);
+
+    _path = m.generate_segment_path(name, *meta);
+
+    auto parsed_name = parse_segment_name(name);
+    vassert(parsed_name, "Can't parse segment name, name: {}", name);
+    _term = parsed_name->term;
+
+    _base_rp_offset = meta->base_offset;
```

Comment:
This is subjective but I don't like coupling via references because we need to think about lifetimes etc. Having just the needed data as owned fields is simpler and more reliable.